book/workflow.rst

```diff
@@ -0,0 +1,236 @@
+
+In this example, the entity can have three states: ``start_call`` (which was configured as the
+start step above), ``start_conversation`` and ``end_call``. The ``end_call`` step is marked as
+final which means that the workflow terminates when this step is reached.
+
+The ``allowed_transitions`` defines which :ref:`transitions <book-workflow-transitions>` can be
+applied in a step.
+
+Workflow Attributes
+...................
+
+During the transition from one step to another step, the user can modify attributes of the
+underlying entity. All attributes that can be modified in any of the transitions must be enabled
+with the ``attributes`` option:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            attributes:
+                phone_call:
+                    label: Phone Call
+                    type: entity
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneCall
+                call_timeout:
+                    type: integer
+                    label: 'Call Timeout'
+                call_successfull:
+                    type: boolean
+                    label: 'Call Successful'
+                conversation_successful:
+                    type: boolean
+                    label: 'Conversation Successful'
+                conversation_comment:
+                    type: string
+                    label: 'Conversation Comment'
+                conversation_result:
+                    type: string
+                    label: 'Conversation Result'
+                conversation:
+                    type: entity
+                    label: 'Conversation'
+                    options:
+                        class: Acme\Bundle\DemoWorkflowBundle\Entity\PhoneConversation
+
+.. _book-workflow-transitions:
+
+Configuring the Transitions
+...........................
+
+Then, you need to define which transitions are available, to which step they transform the entity
+and which attributes can be modified when applying a transition:
+
+.. code-block:: yaml
+    :linenos:
+
+    workflows:
+        phone_call:
+            # ...
+            transitions:
+                connected:
+                    label: 'Connected'
+                    step_to: start_conversation
+                    transition_definition: connected_definition
+                not_answered:
+                    label: "Not answered"
+                    step_to: end_call
+                    transition_definition: not_answered_definition
+                end_conversation:
+                    label: 'End conversation'
+                    step_to: end_call
+                    transition_definition: end_conversation_definition
+
+Transition Pre-Conditions and Post Actions
+..........................................
+
+.. note::
+
+    You can configure as many workflows as you like, but you can only enable one workflow per
+    entity type at the same time.
```

Comment:
Would be good add here information, that during changing of active workflow for an entity all related information about old workflow (WorklfowItem on code level) will be removed. We have warning message on UI, but would be good to have this information in documentation too.